test-interface/src/main/scala/scala/scalanative/testinterface/TestMain.scala

```diff
@@ -31,53 +31,27 @@ object TestMain {
   final val iPv4Loopback = "127.0.0.1"
   final val iPv6Loopback = "::1"
 
-  private def getFreeBSDLoopbackAddr(): String = {
+  private def setFreeBSDWorkaround(): Unit = {
```

Comment:
The "not allowing IPv4 mapped IPv6 addresses," does not match my memory of what was working for me circa 2023-10-15.   When I take a run, hopefully soon, at fixing
unit-tests for FreeBSD, I will have to refresh my memory.

I believe I was running most unit-tests with an unmodified
TestMain on IPv6.  If that was and continues to be true, this section may need to be re-worked.

Reply:
The fact that you had to introduce `iPv6Loopback` and use it on FreeBSD (even if openjdk8 on FreeBSD *has* dual mode sockets) was exactly because IPv4 mapped IPv6 addresses are forbidden, so using `127.0.0.1` on an `AF_INET6` socket wasn't working.
If you enable them, then on Java8  there isn't anything else to do, but on java 11+ the server will continue to listen only on tcp4, unless `java.net.preferIPv4Stack` is explicitely set to false on sbt. Manually adjusting both settings seems much more complex than automatically  push SN to use an IPv4 socket.